backend/src/controllers/userController.ts

```diff
@@ -30,42 +60,190 @@ export const createUser = asyncHandler(async (req, res, next) => {
     classLevel,
     company,
     shareProfile,
-  } = req.body;
+  } = matchedData(req, { locations: ["body"] });
+
+  // check if the user already exists
+  const foundUser = await User.findOne({
+    $or: [{ _id }, { email }],
+  }).lean();
+
+  if (foundUser) {
+    return next(createHttpError(409, "User already exists."));
+  }
+
+  const newUser = new User({
+    _id,
+    email,
+    name,
+    type,
+    linkedIn,
+    phoneNumber,
+    major,
+    classLevel,
+    company,
+    shareProfile,
+  });
+
+  await newUser.save();
+
+  res.status(201).json(newUser);
 });
 
 // @desc Get user by ID
 // @route GET /api/users/:id
 // @access Private
 export const getUserById = asyncHandler(async (req, res, next) => {
   const { id } = req.params;
+
+  // check if the user exists
+  const foundUser = await User.findById(id);
+  if (!foundUser) {
+    return next(createHttpError(404, "User not found."));
+  }
+
+  let responseData: UserResponse = {
+    _id: foundUser._id,
+    email: foundUser.email,
+    name: foundUser.name,
+    type: foundUser.type,
+  };
+
+  if (foundUser.type.includes("Student")) {
```

Comment:
I think its generally better practice to use `student` instead of `Student` for enums to avoid confusion. And also could you create an actual enum just to be safe? Since JavaScript doesn't have an actual enum, this could be something like `type userType = "student" | "alumni";` or I believe there is some other way to freeze an object and use the JSON object as an enum. But a simple type def should be sufficient in this case.